src/org/parosproxy/paros/extension/option/OptionsParamCertificate.java

```diff
@@ -61,14 +70,80 @@ public OptionsParamCertificate() {
     @Override
     protected void parse() {
 
-        // always turn off client cert
-        setUseClientCert(false);
-        setClientCertLocation("");
+        clientCertCheck();
+        saveClientCertSettings();
         
         allowUnsafeSslRenegotiation = getBoolean(ALLOW_UNSAFE_SSL_RENEGOTIATION, false);
         setAllowUnsafeSslRenegotiationSystemProperty(allowUnsafeSslRenegotiation);
     }
 
+    /**
+     *  Saves the client cert settings if the flag is set explicitly.
+     *  Only works for the CLI currently.
+     */
+    private void saveClientCertSettings(){
+
+        if (getInteger(PERSIST_CLIENT_CERT, 0) > 0){
+            logger.warn("Saving Client Certificate settings: password will be found in config");
+            setUseClientCert(getInteger(USE_CLIENT_CERT, 0) > 0);
+            setClientCertLocation(getString(CLIENT_CERT_LOCATION, ""));
+            setClientCertPassword(getString(CLIENT_CERT_PASSWORD, ""));
+
+        } else {
+            // Default to clear settings
+            setUseClientCert(false);
+            setClientCertLocation("");
+            setClientCertPassword("");
+        }
+    }
+
+    /**
+     * Enables ClientCertificate from -config CLI parameters
+     * Requires location, password and a flag to use client certificate.
+     */
+    private void clientCertCheck(){
+
+        int enableClientCert = getInteger(USE_CLIENT_CERT, 0);
+        String certPath = getString(CLIENT_CERT_LOCATION, "");
+        String certPass = getString(CLIENT_CERT_PASSWORD, "");
+
+        if(enableClientCert!=0 && !certPath.isEmpty() && !certPass.isEmpty()) {
+            try {
+
+                SSLContextManager contextManager = getSSLContextManager();
+                int ksIndex = contextManager.loadPKCS12Certificate(certPath, certPass);
+                contextManager.unlockKey(ksIndex, ksIndex, certPass);
+                contextManager.setDefaultKey(ksIndex, ksIndex);
+
+                setActiveCertificate();
+                setEnableCertificate(true);
+
+                logger.info("Client Certificate enabled from CLI");
+                logger.info("Use -config certificate.persist=1 to save settings");
+
+            } catch (IOException ex) {

```

Comment:
These could be merged and use a more informative message (e.g. "The client certificate was not enabled because of an error."), the type is already logged.